File: LadybugTools_Engine/Python/bh_ladybug/location.py
from __future__ import annotations

from typing import TypeVar

T = TypeVar("T")

from ladybug.location import Location


class BH_Location(Location):
    def __init__(
        self,
        city: str = None,
        state: str = None,
        country: str = None,
        latitude: float = 0,
        longitude: float = 0,
        time_zone: str = None,
        elevation: float = 0,
        station_id: str = None,
        source: str = None,
    ) -> BH_Location:
        super().__init__(
            city,
            state,
            country,
            latitude,
            longitude,
            time_zone,
            elevation,
            station_id,
            source,
        )

    def _type(self) -> str:
        return self.__class__.__name__

File: LadybugTools_Adapter/Convert/Location.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace BH.Adapter.LadybugTools
{
    public static partial class Convert
    {
        public static BH.oM.LadybugTools.Location Location(BH.Adapter.LadybugTools.Location oldObject)
        {
            return new oM.LadybugTools.Location()
            {
                City = oldObject.City,
                State = oldObject.State,
                Country = oldObject.Country,
                Latitude = oldObject.Latitude,
                Longitude = oldObject.Longitude,
                TimeZone = oldObject.TimeZone,
                Elevation = oldObject.Elevation,
                StationId = oldObject.StationId,
                Source = oldObject.Source
            };
        }
    }
}
